src/main/java/org/jboss/msc/service/ServiceContainerImpl.java

```diff
@@ -629,7 +628,6 @@ Executor getExecutor() {
      * @return the registration
      */
     ServiceRegistrationImpl getOrCreateRegistration(final ServiceName name) {
-        final ConcurrentMap<ServiceName, ServiceRegistrationImpl> registry = this.registry;
```

Comment:
Why remove this? The JDK does not optimize multiple reads of the same instance field (even if the field is final).

Reply:
I reverted this change.